lib/clientBuilder.js

```diff
@@ -113,6 +113,11 @@ function OktaAuthBuilder(args) {
     return window.location.hash;
   };
 
+  // This is exposed so we can mock setting window.location.hash in our tests
+  sdk.token.parseFromUrl._setLocationHash = function(hash) {
+    window.location.hash = hash;
```

Comment:
yeah, I feel like the history api would take some cross-browser testing (and may be a huge annoyance on embedded browsers), where the hash change is (more) guaranteed to be compatible and ~won't~ shouldn't refresh the page.